jerry-core/jerryscript.h

```diff
@@ -25,6 +25,12 @@ extern "C"
 {
 #endif /* __cplusplus */
 
+#ifdef __GNUC__
+#define __attr_deprecated___ __attribute__((deprecated))
+#else /* !__GNUC__ */
+/* TODO: for other compilers */
+#define __attr_deprecated___
```

Comment:
Does it make sense to "namespace" this, for example `JERRY_ATTR_DEPRECATED` or `__jerry_attr_deprecated__`?
I can imagine other libraries also defining this and causing conflicts in projects.
At the very least, I would add an `#ifndef` around it.